core/src/main/java/io/smallrye/openapi/runtime/scanner/ResourceParameters.java

```diff
@@ -139,7 +142,10 @@ public void sort(List<Parameter> preferredOrder) {
     }
 
     public List<String> getPathParameterTemplateNames() {
-        return getPathParameterTemplateName(this.pathItemPath, this.operationPath);
+        return pathItemPaths.stream()
+                .flatMap(pathItemPath -> operationPaths.stream()
+                        .flatMap(operationPath -> getPathParameterTemplateName(pathItemPath, operationPath).stream()))
+                .collect(Collectors.toList());
```

Comment:
This may need a `distinct()` before the `collect` since the `pathItemPath` will be processed for each entry in `operationPaths`.

```suggestion
        return pathItemPaths.stream()
                .flatMap(pathItemPath -> operationPaths.stream()
                        .flatMap(operationPath -> getPathParameterTemplateName(pathItemPath, operationPath).stream()))
                .distinct()
                .collect(Collectors.toList());
```